drivers/audio/tlv320dac310x.c

```diff
@@ -110,10 +110,10 @@ static int codec_configure(struct device *dev,
 
 	/* configure reset GPIO */
 	gpio_pin_configure(dev_cfg->gpio_device, dev_cfg->gpio_pin,
-				     dev_cfg->gpio_flags | GPIO_DIR_OUT);
+			   dev_cfg->gpio_flags | GPIO_OUTPUT);
 	/* de-assert reset */
-	gpio_pin_write(dev_cfg->gpio_device, dev_cfg->gpio_pin,
-				 CODEC_RESET_PIN_DEASSERT);
+	gpio_pin_set_raw(dev_cfg->gpio_device, dev_cfg->gpio_pin,
```

Comment:
In the call to `gpio_pin_configure` we provide gpio flags and as such honor active/inactive state. It may be more appropriate to use `gpio_pin_set` here.

Reply:
I am not sure whether the DT source for this codec actually specifies active low/high. I will check. But note that the codec has macros like `CODEC_RESET_PIN_DEASSERT` based on raw access, and that’s why I used set_raw

Reply:
I support it the way it is; this was raised in slack#gpio and https://github.com/zephyrproject-rtos/zephyr/issues/20017#issuecomment-549315497 updated to describe why it may be done this way (**NOTE** at bottom).

Reply:
The one board that is using the codec is `boards/xtensa/intel_s1000_crb/intel_s1000_crb.dts`. It doesn't specify the active flag. However, in case of the reset signal we've been generally honoring pin active / inactive state definition. Giving the board flexibility to route the signal via an inverter. That approach is also consistent with how Linux handles it.

If we decided to use `gpio_pin_set` we would need to update `CODEC_RESET_PIN_DEASSERT` macros and `boards/xtensa/intel_s1000_crb/intel_s1000_crb.dts` file.

Reply:
So in this particular case, given that `CODEC_RESET_PIN_ASSERT` is 0  and  `CODEC_RESET_PIN_DEASSERT` is 1 this should be `ACTIVE_LOW`, correct? If I make that change I will simply remove the macros and use `gpio_pin_set(dev_cfg->gpio_device, dev_cfg->gpio_pin, 0)`. Does that make sense?

Reply:
Yes, the reset pin should be marked as active low in dts. Removing the macros is a good idea. We could actually skip `gpio_pin_set` altogether and pass `GPIO_OUTPUT_INACTIVE` to `gpio_pin_configure` function.

Reply:
OK, you convinced me on this one @mnkp. Please take a look (@pabigot too please, if you find a moment).